js/bitmex.js

```diff
@@ -256,139 +421,155 @@ module.exports = class bitmex extends Exchange {
+        //        "fairBasis": "-3.83",
+        //        "fairPrice": "19132.44",
         //        "markMethod": "FairPrice",
-        //        "markPrice": 143.811,
+        //        "markPrice": "19132.44",
         //        "indicativeTaxRate": null,
-        //        "indicativeSettlePrice": 143.807,
+        //        "indicativeSettlePrice": "19136.27",
         //        "optionUnderlyingPrice": null,
         //        "settledPriceAdjustmentRate": null,
         //        "settledPrice": null,
-        //        "timestamp": "2022-01-14T17:49:55.000Z"
-        //    }
+        //        "timestamp": "2022-10-16T14:52:20.000Z"
+        //      },
+        //     ]
         //
         const result = [];
         for (let i = 0; i < response.length; i++) {
             const market = response[i];
             const id = this.safeString (market, 'symbol');
-            const baseId = this.safeString (market, 'underlying');
-            const quoteId = this.safeString (market, 'quoteCurrency');
-            const settleId = this.safeString (market, 'settlCurrency', '');
+            const typ = this.safeString (market, 'typ'); // type definitions at: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            const baseId = this.currencyIdFromAssetName (this.safeString (market, 'underlying'));
+            const quoteId = this.currencyIdFromAssetName (this.safeString (market, 'quoteCurrency'));
+            const settleId = this.currencyIdFromAssetName (this.safeString (market, 'settlCurrency', ''));
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
             const settle = this.safeCurrencyCode (settleId);
-            const basequote = baseId + quoteId;
-            const swap = (id === basequote);
             // 'positionCurrency' may be empty ("", as Bitmex currently returns for ETHUSD)
             // so let's take the settlCurrency first and then adjust if needed
             let type = undefined;
+            let spot = false;
             let future = false;
+            let swap = false;
             let prediction = false;
             let index = false;
-            let symbol = base + '/' + quote + ':' + settle;
+            let symbol = undefined;
             const expiryDatetime = this.safeString (market, 'expiry');
             const expiry = this.parse8601 (expiryDatetime);
             const inverse = this.safeValue (market, 'isInverse');
             const status = this.safeString (market, 'state');
             let active = status !== 'Unlisted';
-            if (swap) {
+            // types defined here: https://www.bitmex.com/api/explorer/#!/Instrument/Instrument_get
+            if (typ === 'FFWCSX' || typ === 'FFWCSF') {
                 type = 'swap';
+                swap = true;
+                symbol = base + '/' + quote + ':' + settle;
+            } else if (typ === 'IFXXXP') {
+                type = 'spot';
```

Comment:
@ttodua for some reason "SOL/USDT" does not seem to be loaded:
https://www.bitmex.com/spot/SOL_USDT
![image](https://user-images.githubusercontent.com/43336371/208091202-286cb6ac-24c4-4c0b-b734-2804b098d400.png)

Reply:
what a good catch. bitmex has very weird response. i.e. in response of currencies, for SOL , the assetname is "SOL" and id is "LAMp". for other symbols, they were almost same and I couldnt notice such thing might exist. I've made the correct definition of market symbols (and their baseId/quoteId). now it is correct I am confident.